app/assets/javascripts/discourse/app/lib/session-store.js

```diff
@@ -0,0 +1,128 @@
+import { isTesting } from "discourse/lib/environment";
+
+const TEST_KEY_PREFIX = "__test_";
+
+/**
+ * @type {Storage}
+ */
+let safeSessionStorage;
+
+try {
+  safeSessionStorage = sessionStorage;
+  if (sessionStorage["disableSessionStorage"] === "true") {
+    safeSessionStorage = null;
+  } else {
+    // makes sure we can write to the session storage
+    safeSessionStorage["safeSessionStorage"] = true;
+  }
+} catch {
+  // session storage disabled
+  safeSessionStorage = null;
+}
```

Comment:
Is there any known/documented situation where this would fail? If not, I would just remove it.